### ISSUE TITLE
Bug: AST changed with strings containing \r\n

### ISSUE THREAD

**Issue by Julow** [ocaml-ppx/ocamlformat#2601]
A reproduction can be found in `test-extra/code/ocaml/testsuite/tests/lexing/newlines.ml`. This is new since 0.26.2.

**Comment by Julow** [comment]
This bug is noticeable since https://github.com/ocaml-ppx/ocamlformat/pull/2557

**Comment by hhugo** [comment]
The issue comes from https://github.com/ocaml/ocaml/pull/12514, `lib/Literal_lexer.mll` needs to be updated accordingly.

**Comment by hhugo** [comment]
The diff below should be enough to fix the issue with OCaml 5.2 and above - but will cause issues with older version. 


```diff
$ git diff
diff --git a/lib/Literal_lexer.mll b/lib/Literal_lexer.mll
index 95ed7eec..e05ec78e 100644
--- a/lib/Literal_lexer.mll
+++ b/lib/Literal_lexer.mll
@@ -81,7 +81,9 @@ and string_aux mode = parse
         string_aux mode lexbuf
       }
   | newline
-      { store_string (Lexing.lexeme lexbuf);
+      { (match Lexing.lexeme lexbuf with
+        | "\n" -> store_string "\n"
+        | s -> store_string (String.sub s 1 (String.length s - 1)));
         string_aux mode lexbuf }
   | eof
       { raise Parse_error }
```
